xdsl/dialects/mpi.py

```diff
@@ -90,26 +82,33 @@ class ISend(MPIBaseOp):
 
     ## Our Abstraction:
 
-        - We Summarize buf, count and datatype by using memrefs
-        - We assume that tag is compile-time constant
-        - We omit the possibility of using multiple communicators
+        - We omit the possibility of using multiple communicators, defaulting
+          to MPI_COMM_WORLD
     """
 
     name = 'mpi.isend'
 
-    buffer: Annotated[Operand, MemRefType[AnyNumericType]]
+    buffer: Annotated[Operand, AnyAttr()]
```

Comment:
```suggestion
    buffer: Operand
```
should be enough, or
`buffer: Annotated[Operand, Attribute]` if you prefer keeping the `Annotated`